internal/kafka/internal/api/dbapi/kafka_request_types.go

```diff
@@ -2,13 +2,15 @@ package dbapi
 
 import (
 	"encoding/json"
+	"time"
 
 	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/api"
 	"gorm.io/gorm"
 )
 
 type KafkaRequest struct {
 	api.Meta
+	ExpiresAt                        *time.Time
```

Comment:
Do we need to persist the information in the database?

the question is more or less: do we want to record the expiration time of an instance and make sure that any future change of the configuration value does not impact the value set for already existing instances?

Reply:
I'd say so. It is also useful information to consumers of the get/list endpoint

Reply:
> Can you expand on this? How easier?

You only need to compare the field value and the current time

Reply:
> > Can you expand on this? How easier?
> 
> You only need to compare the field value and the current time stored in the kafka request

This is more or less what is there current but doing the comparison on the fly as well. The lifespan is passed in as an argument and it is used to perform the db update. If no expires_at is stored then the logic requires no change other than continuing to pass in the lifespan duration. If the expires_at is stored then the deprovision logic will need to be revisited. But as far as the comparison goes, I do find the two ways of doing things at the same level of understandability.

Reply:
> > The question was specifically about persisting the information in the database. The API can still return the expires_at information without persisting it in the database. It'll return it on the fly on the presentation layer.
> 
> It is a possible option, although as you said the implication is that then if we change the values in the supported kafka instance types it would imply a change of the expiration time in both the returned responses and in the expiration logic (when implemented). 

This is what I’d like to get clarity on: what’s the expected behaviour? At the moment, the behavious is - if the lifespan flag is changed, the developer instance (old and existing ones) will expires at a time equals to “created_at + new_lifespan_duration”. 
if we store the value and updated the deprovision logic based on to use the expires_at, we are changing the current behaviour, is this desirable or a side effect of work in the PR?

> Aside from that, if for some reason in the future we let POST/PATCH specify an expire time that would not work

In the future, if it is needed, it can be added.

Reply:
> This is what I’d like to get clarity on: what’s the expected behaviour? At the moment, the behavious is - if the lifespan flag is changed, the developer instance (old and existing ones) will expires at a time equals to “created_at + new_lifespan_duration”.
if we store the value and updated the deprovision logic based on to use the expires_at, we are changing the current behaviour, is this desirable or a side effect of work in the PR?

I guess that is not really defined/decided. Both options seem acceptable to me.
I would like other thoughts about this from @JameelB.

> In the future, if it is needed, it can be added.

That is correct.

Reply:
If we want to keep the current behavior, where changing KAFKAS_LIFE_SPAN value impacts all already pre-existing kafka instances lifetime then the approach would be not to persist the value in the DB

Reply:
Good point @machi1990. For me both options are reasonable as well. +1 with what @miguelsorianod said, we haven't really discussed it. Is the current behaviour the actual desired behaviour? 

I think adding the expires_at in the db would definitely be more future proof. It will enable us to allow users to request an extension for short-lived Kafkas. 

We don't really change the lifespan duration that much. With the current behaviour though, if i was the user of a Kafka and was told I had an x amount of hours left at the start then it was changed to a shorter time limit and the Kafka expired without me knowing, i'd find it confusing as to why that Kafka disappeared all of the sudden. 

I guess question is, if we told the user at the start they have x amount of hours on their Kafka, do we need to ensure that their Kafka stays available for that period of time?

Reply:
Can’t `expire_at` be used only for the eventual trial extend case, while deriving it for all other instances?

Reply:
@riccardo-forina `expires_at` is only relevant for trial/developer instances. standard instances don't expire.

Reply:
Decision. Keep the behavior we have for now, and we return expires_at always if the instance has a lifespan:
1. Do not store it in the DB
2. If a kafka size has a lifespan set we ALWAYS set the expires_at
3. If a kafka size does not have a lifespan we never set the expires_at
4. expires_at can't be set by a user. If that is needed then we will change the behavior. The risk here is that the semantics of the expires_at field might change in that case which might be potentially breaking behavior
5. This means if we change the lifespan of an instance size the already existing instances are impacted. This is what is already happening